Vhar 7429 alihankintabonus

rckopteri opened the pull request:
2022 ja sitä myöhemmin alkaville hoitokausille kaikki bonus sivulta tehdyt alihankintabonukset näytetään ikäänkuin ne olisivat rahavarauksia.
Alihankintabonukset siirretään MHU Ylläpito toimenpideinstanssin alle.
Muutoksia on tehty myös maksueriin, laskutusyhteevetoon, kustannusten seurantaan ja yksikkötesteihin.

rckopteri committed:
Muodosta bonusten haku uusiksi kustannusten seurannassa

Jotta alihankintabonus merkitään tilaajan rahavaraukseksi

rckopteri committed:
Päivitä testit vastaamaan muuttunutta toteutusta

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Siirrä alihankintabonus rahavaraukseksi mhu_laskutusyhteenvedossa

rckopteri committed:
Siirrä alihankintabonus rahavaraukseksi mhu_laskutusyhteenvedossa

rckopteri committed:
Näytä raportti-ui:lla alihankintabonus rahavarauksena

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Eriytä rahavaraukset näkymään tavoitehinnassa

rckopteri committed:
Ota alihankintabonuksen siirto rahavarauksiin käyttöön vain 2022 alkaville ja myöhemmin

rckopteri committed:
Päivitä testit ja testiaineisto testaamaan alihankintabonuksen siirtoa rahavarauksiin

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Päivitä testit vastaamaan uutta toteutusta ja uusia testiaineistoja

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Poista turha logitus

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

# Conflicts:
#	src/clj/harja/palvelin/raportointi/raportit/laskutusyhteenveto_tuotekohtainen.clj

rckopteri committed:
Merkitse bonuslomakkeella alihankintabonus MHU Yllapitoon

rckopteri committed:
Himmennä tehtäväryhmäteksti bonuslomakkeella

K1ll3rF0x committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Poista T3 - Tilaajan rahavaraus kulujen lisäämissivulta

rckopteri committed:
Korjaa alihankintabonuksen näyttäminen

Tilaajan rahavarauksena ja negatiivisena lukuna positiiviseksi

rckopteri committed:
Näytaä alihankintabonus tilaajan rahavarauksena

Mikäli on hoitokauden alkuvuosi 2022 tai myöhemmin

rckopteri committed:
Muuta alihankintabonus MHU Ylläpidolle 1.10.2022 jälkeen

rckopteri committed:
Lisää maksuerätesteihin alihankintabonuksen testaus

rckopteri committed:
Lisää työmaaraportti laskutusyhteevetotestiin alihankintabonuksen testausta

rckopteri committed:
Lisää helpper funkkareita testeihin

rckopteri committed:
Päivitä työmaaversio ja tuoteversio laskutusyhteenvedosta

Lisää alihankintabonuksen käsittelyjä 1.10.2022 jälkeen

rckopteri committed:
Päivitä testidata vastaamaan paremmin oikeaa tietoa

alihankintabonuksen muodostumisesta

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Poista testiaineistosta testejä sekottavia alihankintabonuksia

Ja korjaa rikkoutuneet yksikkötestit.

rckopteri committed:
Korjaa tuotekohtainen laskutusyhteenvetotesti

solita-staskila commented on tietokanta/src/main/resources/db/migration/R__Laskutusyhteenveto_tyomaakokous.sql:
```diff
@@ -35,6 +35,8 @@ CREATE TYPE LY_RAPORTTI_TYOMAAKOKOUS_TULOS AS
     akilliset_val_aika_yht                NUMERIC,
     vahingot_hoitokausi_yht               NUMERIC,
     vahingot_val_aika_yht                 NUMERIC,
+    alihankintabonus_hoitokausi_yht       NUMERIC,
```
T̶a̶r̶v̶i̶t̶a̶a̶n̶k̶o̶ ̶n̶ä̶i̶t̶ä̶?̶

Nimetäisiinkö nämä uudelleen? -> 
alihank_bon_hoitokausi_yht          NUMERIC; 
alihank_bon_val_aika_yht              NUMERIC;

solita-staskila commented on src/clj/harja/palvelin/raportointi/raportit/laskutusyhteenveto_tyomaa.clj:
```diff
@@ -44,7 +44,17 @@
+                                (= "Rahavaraukset" otsikko))
+                              [(rivi-taulukolle data kyseessa-kk-vali? "Äkilliset hoitotyöt" :akilliset_hoitokausi_yht :akilliset_val_aika_yht false)
+                               (rivi-taulukolle data kyseessa-kk-vali? "Vahinkojen korjaukset" :vahingot_hoitokausi_yht :vahingot_val_aika_yht false)
+                               (rivi-taulukolle data kyseessa-kk-vali? "Tilaajan rahavaraus (alihankintasopimusten maksuehtobonukset)" :alihankintabonus_hoitokausi_yht :alihankintabonus_val_aika_yht false)]
```
Palauttaa tyhjät arvot "" mikäli arvoja ei ole, voisiko palauttaa 0.00 kuten kaikki muutkin rivit yhteenvedossa?

solita-staskila reviewed (approved):
En nähnyt mitään kummempaa, koodi siisti, pyöräyttelin muutamat funktiot ja näyttää toimivan 👍

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Päivitä laskutusyhteenveto käyttämään laskutuskuukautta

Kun haetaan erilliskustannuksista.
Päivitä samalla testiaineistoihin laskutuskuukausi

rckopteri committed:
Päivitä tuotannon erilliskustannuksille laskutuskuukausi, jos se puuttuu

rckopteri committed:
Näytä infolaatikko T3 tehtäväryhmästä.

Ja jäsennä radiogrouppien iffittelyä hieman

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Ota laskutuskuukausi erilliskustannuksissa huomioon myös muualla ja raporteissa

rckopteri committed:
Lisää vielä puuttuvat laskutuskuukaudet hakuihin

rckopteri committed:
Varmista, että testit toimii

erilliskustannuksen laskutuskuukauden lisäämisen jälkeen

rckopteri committed:
Piilota infolaatikko 1.1.2025 alussa

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

# Conflicts:
#	test/clj/harja/kyselyt/maksuerat_test.clj
#	tietokanta/src/main/resources/db/migration/V1_1040__.sql

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Korjaa päivämäärä oikeaksi

rckopteri committed:
Vapauta T3 tehtäväryhmä

Tietokannasta, mutta estä sen käyttö käyttöliittymästä

rckopteri committed:
Nimeä käsittelykuukausi -> laskutuskuukausi

rckopteri committed:
Varmista, että testit toimii

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_1043__.sql

rckopteri committed:
Päivitä testit vastaamaan toteutusta

Ei ole enää kiellettyä hakea Tilaajan rahavaraus (T3) tehtäväryhmää tietokannasta.

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Poista kuukausikohtainen filtteri

 sanktiot ja bonukset näkymästä

rckopteri committed:
Säädä filttereiden ja lisää napin

sijainti, kun osa filttereistä poistui

rckopteri committed:
Poista erilliskustannuksista Hoidon johto

toimenpideinstanssi vaatimus. Koska alihankintabonukset kuuluvat 1.10.2022 jälkeen ylläpito toimenpideinstanssille.

rckopteri committed:
Varmista, että alihankintabonuksen

lisääminen aiheuttaa toimenpideinstanssimuutoksen, koska ne kuuluu 1.10.2022 alkaen ylläpito -toimenpideinstanssille.

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Muutetaan Tehtävyhrmä -tekstin kontrasti paremmaksi

rckopteri committed:
Lisää Laskutuskuukausi sanktiot ja bonukset listaan

Samalla on pakko säätää muiden kolumnien leveyttä

rckopteri committed:
Korjaa toimenpideinstanssi valikon toiminta hoitourakoille

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_1044__.sql

rckopteri committed:
Korjaa typo infotekstistä

rckopteri committed:
Estä laskutuskuukauden muutos, jos bonus on jo tallennettu kantaan

rckopteri committed:
Formatoi laskutuskuukausi

bonukset&sanktiot listaukseen samaan tapaan kuin lomakkeella

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_1045__.sql

rckopteri committed:
Muuta sprokin parametrien nimiä ja korjaa yksikkötesti

rckopteri committed:
Merge branch 'develop' into VHAR-7429-alihankintabonus

# Conflicts:
#	src/cljs/harja/views/urakka/kulut/kulut.cljs
#	tietokanta/src/main/resources/db/migration/V1_1048__.sql

rckopteri committed:
Korjaa git migraatio konfliktiongelma

rckopteri committed:
Korjaa yksikkötestit
